num/src/smallint.rs

```diff
@@ -103,6 +105,48 @@ macro_rules! construct_smallint {
             }
         }
 
+        impl DivRem for $ty {
+            // divmod like operation, returns (quotient, remainder)
+            #[inline]
+            fn div_rem(self, other: Self) -> (Self, Self) {
+                //quotient and remainder will always be smaller than self so they're going to be in bounds
+                let quotient = self / other;
```

Comment:
I think we need to check that the operation does not zero divide - or add an information to the doc comment that it will panic in this case - and have `div_rem_checked` version as for other arithmetic operations which will return `Option<(Self, Self)>`

Reply:
In BigInt there are an assert, maybe I can include the same:
assert(other != 0).
And div_rem_checked would be another method for the trait?